### PR TITLE
refactor: Get rid of JSON in parameter_table

### PR DESCRIPTION
**Pull request by akashin**
This PR simplifies parameter parsing code by removing JSON usage and instead directly storing parameter values as YAML. Functionally, the code should behave the same for the users of `ParameterTable`.

This is a part of refactoring for https://github.com/near/nearcore/issues/8264.

I'm still pondering using a more specialized enum type to store parameter values, but not yet sure how the implementation will look like, so delaying this until I start implementing parameter weights in config.

### CODE REVIEW

File: core/primitives/src/runtime/parameter_table.rs [hunk 1]
```diff
@@ -149,23 +146,23 @@ impl ParameterTable {
...
             let mut key: &'static str = param.into();
             key = key.strip_prefix(remove_prefix).unwrap_or(key);
             if let Some(value) = self.get(*param) {
-                json.insert(key.to_owned(), value.clone());
+                yaml.insert(key.into(), value.clone());
             }
         }
-        json.into()
+        yaml.into()
     }
```
Comment (jakmeier) [inline core/primitives/src/runtime/parameter_table.rs:163]: BTW: This whole construct to group parameters by a prefix should probably be revisited, too. When we wanted to make it a linear list of parameters, this was the least ugly work-around we found. (But not part of this PR.)

File: core/primitives/src/runtime/parameter_table.rs [hunk 2]
```diff
@@ -185,36 +182,49 @@ impl ParameterTable {
...
-            "send_not_sir": self.get(format!("{key}_send_not_sir").parse().unwrap()),
-            "execution": self.get(format!("{key}_execution").parse().unwrap()),
-        })
+    fn fee_yaml(&self, key: FeeParameter) -> serde_yaml::Value {
+        serde_yaml::to_value(BTreeMap::from([
+            ("send_sir", self.get(format!("{key}_send_sir").parse().unwrap())),
+            ("send_not_sir", self.get(format!("{key}_send_not_sir").parse().unwrap())),
+            ("execution", self.get(format!("{key}_execution").parse().unwrap())),
+        ]))
+        .expect("failed to construct fee yaml")
```
Comment (jakmeier) [inline core/primitives/src/runtime/parameter_table.rs:227]: we may also want to reconsider this construct, but not the highest priority on my list

**Review by jakmeier — APPROVED** [review]
LGTM

> I'm still pondering using a more specialized enum type to store parameter values, but not yet sure how the implementation will look like, so delaying this until I start implementing parameter weights in config.

:+1:

File: core/primitives/src/runtime/parameter_table.rs (hunk 1 above)
Comment (akashin) [reply core/primitives/src/runtime/parameter_table.rs:163]: I agree that we should simplify this, especially given that there is only one usage to construct `VMLimitConfig` above. I haven't found a simple way to do this that doesn't require macro programming (as this is basically what serde::Deserialize that we use now relies on), but I'll keep looking.

File: core/primitives/src/runtime/parameter_table.rs (hunk 2 above)
Comment (akashin) [reply core/primitives/src/runtime/parameter_table.rs:227]: For this one, I plan to parse and store it as a structured parameter
```rust
struct Fee {
  send_sir: Gas,
  send_not_sir: Gas,
  execution: Gas,
}
```
Will working on this and weights representation in the next PR.

The main challenge here will be to keep the diff-config merging code relatively simple, but I think that should be doable.
